### PR TITLE
TM-1323 New on page condition

### PR DESCRIPTION
**Pull request by marnits**

### CODE REVIEW

File: src/SevenTag/Api/TriggerBundle/TriggerType/ViewPreparator/ConditionViewPreparator.php [hunk 1]
```diff
@@ -111,14 +98,45 @@ private function mapVariablesToViewRepresentation(ArrayCollection $variables)
...
+        $actions = [];
+
+        /** @var VariableTypeInterface $variableType */
+        foreach ($type->getAllowedVariables() as $variableType) {
+            if ($variableType->getName() === $variable->getName()) {
+                foreach ($variableType->getAllowedActions() as $actionType => $action) {
+                    $actions[$actionType] = [
+                        'type' => $actionType,
+                        'name' => $action['name'],
+                        'unit' => $action['unit']
```
Comment (MarcinMat) [inline src/SevenTag/Api/TriggerBundle/TriggerType/ViewPreparator/ConditionViewPreparator.php]: What's this?

Comment (marnits) [inline src/SevenTag/Api/TriggerBundle/TriggerType/ViewPreparator/ConditionViewPreparator.php]: What is what?

We are looking for variable (by name), and then building array with actions (if exists - if not, get default).

File: src/admin-panel/app/modules/condition/tests/services/Condition.spec.js [hunk 2]
```diff
@@ -25,249 +25,151 @@ describe('Unit: Condition service', () => {
...
+                                type: 'contains',
+                                name: 'Contains'
+                            },
+                            {
+                                type: 'not_contains',
+                                name: 'Not contains'
+                            },
+                            {
+                                type: 'start_with',
+                                name: 'Start with'
```
Comment (MarcinMat) [inline src/admin-panel/app/modules/condition/tests/services/Condition.spec.js]: Maybe 'Starts with' ?

File: tests/containerjs/unit/collector.spec.js [hunk 3]
```diff
@@ -53,6 +54,7 @@ describe('Unit: collector module', function () {
 
 
             expect(documentCollector(testVariable)).toBe('something');
+            expect(documentCollector(testVariable)).toBe('something');
```
Comment (MarcinMat) [inline tests/containerjs/unit/collector.spec.js]: Is this a duplicate?

File: tests/containerjs/unit/collector.spec.js [hunk 4]
```diff
@@ -69,4 +71,25 @@ describe('Unit: collector module', function () {
             expect(documentCollector(testArrayLikeVariable)).not.toBeDefined();
         });
     });
+
+    describe('document collector', function () {
+        beforeEach(function () {
+            $cookie.add('stg_session_cookie1', 'test1');
+            $cookie.add('stg_session_cookie2', 'test2');
+            $cookie.add('stg_test_cokie3', 'test3');
```
Comment (MarcinMat) [inline tests/containerjs/unit/collector.spec.js]: `andCallFake` on `$cookie.get` should be used instead adding cookies. Also, none of this method is mocked (should be), there is no reason to call real implementation that affects the cookies.

File: src/containerjs/visitor/condition/duringTheLastVisitor.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+'use strict';
+
+(function(sevenTag, MODULE_NAME) {
+    var DuringTheLastVisitor = function($visitHandler) {
```
Comment (MarcinMat) [inline src/containerjs/visitor/condition/duringTheLastVisitor.js]: I'm not sure about the name. `DuringTheLastVisitor` is not descriptive enough. What does 'last visitor' mean?

Comment (marnits) [inline src/containerjs/visitor/condition/duringTheLastVisitor.js]: This is the convention of all ours visitors

File: src/containerjs/visitor/condition/duringTheLastVisitor.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            }
+
+            return visitNotDuringTheLast(target, variables);
+        };
+
+        var visitDuringTheLast = function(target, variables) {
+            return $visitHandler.handle(target.value, variables[target.variable]) === -1;
+        };
+
+        var visitNotDuringTheLast = function(target, variables) {
```
Comment (MarcinMat) [inline src/containerjs/visitor/condition/duringTheLastVisitor.js]: Again, name of the function.

File: src/containerjs/cookie/cookie.js [hunk 7]
```diff
@@ -70,6 +70,34 @@
 
         /**
          * @param name
+         * @returns {*}
+         */
+        this.getByRegex = function (name) {
```
Comment (MarcinMat) [inline src/containerjs/cookie/cookie.js]: A loot of duplication with `get()`. Make some internal utils function that accepts `cookieFilter`, that either compares by name or by regex.

Also parameter name is `name`. Should be regexPattern or something like this.

File: src/containerjs/visitor/condition/duringTheLastVisitor.js (hunk 6 above)
Comment (marnits) [inline src/containerjs/visitor/condition/duringTheLastVisitor.js]: again, its the convention

**Pull request merged**